app/controllers/surveys_controller.rb

```diff
@@ -0,0 +1,91 @@
+get '/surveys/:survey_id' do
+  @survey = Survey.find(params[:survey_id])
+  if logged_in?
+    if @survey.user_answers.any?{|user_answer| user_answer.user_id == current_user.id}
```

Comment:
Too much logic in your controller.

What's the question you're asking?

"Are there any answers from this user on this survey?"

What does that look like as a method?

```ruby
some_model.has_been_taken_by?(current_user)
```

Whose job is it to work this out?

The Survey.

```ruby
survey.has_been_taken_by?(current_user)
```